commons/src/test/java/com/powsybl/commons/extensions/ExtensionAdderProvidersTest.java

```diff
@@ -0,0 +1,355 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.commons.extensions;
+
+import static org.junit.Assert.*;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.junit.Test;
+
+import com.google.auto.service.AutoService;
+import com.powsybl.commons.PowsyblException;
+
+public class ExtensionAdderProvidersTest {
+
+    private static interface SimpleExtendable extends Extendable<SimpleExtendable> {
+    }
+
+    private static class SimpleExtendableImpl2
+            extends AbstractExtendable<SimpleExtendable> implements SimpleExtendable {
+        @Override
+        public String getImplementationName() {
+            return "Custom";
+        }
+    }
+
+    private static interface SimpleExtension extends Extension<SimpleExtendable> {
+        default String getName() {
+            return "SimpleExtension";
+        }
+    }
+
+    private static class SimpleExtensionImpl extends AbstractExtension<SimpleExtendable>
+            implements SimpleExtension {
+    }
+
+    private static class SimpleExtensionImpl2 extends AbstractExtension<SimpleExtendable>
+            implements SimpleExtension {
+    }
+
+    private static interface SimpleExtensionAdder extends ExtensionAdder<SimpleExtendable, SimpleExtension> {
+        default Class<SimpleExtension> getExtensionClass() {
```

Comment:
It's dirty but I assume we can write dirty code in tests... Default should not be used to avoid duplications